### ISSUE TITLE
Different loss function in ranking model in xgboost

### ISSUE THREAD

**Issue by dengjn** [dmlc/xgboost#2562]
I tried both rank:pairwise and rank:ndcg as loss function in XGboost, and found rank:pairwise is always better than rank:ndcg. rank:ndcg is based on lambdaMART which should have better performance than rank:pairwise. Can anyone  explain to me the difference between the two loss function?

thanks,
jd

**Comment by pommedeterresautee** [comment]
rank:ndcg is based on LambdaRank.
rank:pairwise is based on LambdaMart.

In general LambdaMart performs better than LambdaRank.
In most cases, pairwise was working better than listwise approach, but it seems than with DL this is changing.

For more info:
https://www.microsoft.com/en-us/research/publication/from-ranknet-to-lambdarank-to-lambdamart-an-overview/
